lib/command.js

```diff
@@ -106,22 +129,37 @@ Command.prototype.getKeys = function () {
           if (directive === 'GET') {
             i += 1;
             if (this.args[i] !== '#') {
+              if (transform) {
+                this.args[i] = transform(this.args[i]);
+              }
               this._keys.push(this.getKeyPart(this.args[i]));
             }
           } else if (directive === 'BY') {
             i += 1;
+            if (transform) {
+              this.args[i] = transform(this.args[i]);
+            }
             this._keys.push(this.getKeyPart(this.args[i]));
           } else if (directive === 'STORE') {
             i += 1;
+            if (transform) {
+              this.args[i] = transform(this.args[i]);
+            }
             this._keys.push(this.args[i]);
           }
         }
         break;
       case 'zunionstore':
       case 'zinterstore':
+        if (transform) {
+          this.args[0] = transform(this.args[0]);
+        }
         this._keys.push(this.args[0]);
         keyStop = parseInt(this.args[1], 10) + 2;
         for (i = 2; i < keyStop; ++i) {
+          if (transform) {
```

Comment:
Yes, and I prefer the latter way to create a function for `transform` and `_keys.push`:
```javascript
function (arg, keys)
  keys.push(transform ? transform(arg) : arg);
}
```
However I'm not sure whether that would impact the performance.